engines/router/missionctl/log/resultlog/kafka.go

```diff
@@ -130,25 +126,101 @@ func (l *KafkaLogger) write(turLogEntry *TuringResultLogEntry) error {
 	return nil
 }
 
-// newKafkaLogEntry converts a given TuringResultLogEntry to a kafkaLogEntry, to
-// be written to a Kafka topic
-func newKafkaLogEntry(routerVersion string, e *TuringResultLogEntry) (*kafkaLogEntry, error) {
+// newJSONKafkaLogEntry converts a given TuringResultLogEntry to arbitrary struct with the
+// required fields and marshals it into a byte array, for writing to a Kafka topic
+func newJSONKafkaLogEntry(routerVersion string, e *TuringResultLogEntry) (messageBytes []byte, err error) {
 	// Get Turing request id
-	turingReqID, err := turingctx.GetRequestID(*e.ctx)
+	var turingReqID string
+	turingReqID, err = turingctx.GetRequestID(*e.ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	logEntry := &kafkaLogEntry{
-		RouterVersion: routerVersion,
-		TuringReqID:   turingReqID,
-		Timestamp:     e.timestamp.Format(time.RFC3339Nano),
-		Request:       e.request,
-		Experiment:    getTuringResponseOrNil(e.responses, ResultLogKeys.Experiment),
-		Enricher:      getTuringResponseOrNil(e.responses, ResultLogKeys.Enricher),
-		Router:        getTuringResponseOrNil(e.responses, ResultLogKeys.Router),
-		Ensembler:     getTuringResponseOrNil(e.responses, ResultLogKeys.Ensembler),
+	kafkaMessage := &struct {
+		RouterVersion  string            `json:"router_version"`
+		TuringReqID    string            `json:"turing_req_id"`
+		EventTimestamp string            `json:"event_timestamp"`
+		Request        requestLogEntry   `json:"request"`
+		Experiment     *responseLogEntry `json:"experiment,omitempty"`
+		Enricher       *responseLogEntry `json:"enricher,omitempty"`
+		Router         *responseLogEntry `json:"router,omitempty"`
+		Ensembler      *responseLogEntry `json:"ensembler,omitempty"`
+	}{
+		RouterVersion:  routerVersion,
+		TuringReqID:    turingReqID,
+		EventTimestamp: e.timestamp.Format(time.RFC3339Nano),
```

Comment:
This was `Ts` earlier, to match the BQ schema. But changing this to `EventTimestamp` to match the internal Kafka message convention.

Reply:
I think it's good if the default behaviour of our various log exporters to different sinks (Kafka, BigQuery, Fluentd) is consistent with the schema of our log message.

If we are to use this `TuringResultLog.proto` as the schema of our log message, 
```proto
message TuringResultLogMessage {
  string turing_req_id = 1;
  google.protobuf.Timestamp event_timestamp = 2;
  string router_version = 3;
  Request request = 4;
  Response experiment = 5;
  Response enricher = 6;
  Response router = 7;
  Response ensembler = 8;
}
```

I think it is reasonable to expect that if someone choose BigQuery as the LogSink, there should be a column called `event_timestamp` rather than having it automatically renamed to `ts`.

Reply:
I've added a card to track this internally. We need to handle this update for existing users of BQ logging.